src/model/ClassRegistry.js

```diff
@@ -1,535 +1,575 @@
-    'SurgicalApproach': require('./shr/procedure/SurgicalApproach').default,
-    'Access': require('./shr/procedure/Access').default,
-    'UsingDevice': require('./shr/procedure/UsingDevice').default,
-    'UsingDeviceCode': require('./shr/procedure/UsingDeviceCode').default,
-    'UsingAccessDevice': require('./shr/procedure/UsingAccessDevice').default,
-    'UsingAccessDeviceCode': require('./shr/procedure/UsingAccessDeviceCode').default,
-    'IndirectDevice': require('./shr/procedure/IndirectDevice').default,
-    'IndirectDeviceCode': require('./shr/procedure/IndirectDeviceCode').default,
-    'SpecimenCollectionPerformed': require('./shr/procedure/SpecimenCollectionPerformed').default,
-    'AmountOrSize': require('./shr/procedure/AmountOrSize').default,
-    'DiagnosticImagingPerformed': require('./shr/procedure/DiagnosticImagingPerformed').default,
-  },
-  'shr.research': {
-    'Study': require('./shr/research/Study').default,
-    'Enrollment': require('./shr/research/Enrollment').default,
-    'Sponsor': require('./shr/research/Sponsor').default,
-    'Jurisdiction': require('./shr/research/Jurisdiction').default,
-    'PrincipalInvestigator': require('./shr/research/PrincipalInvestigator').default,
-    'TerminationReason': require('./shr/research/TerminationReason').default,
-    'StudyArm': require('./shr/research/StudyArm').default,
-    'ResearchSubject': require('./shr/research/ResearchSubject').default,
-  },
-  'tumor': {
-    'TumorPresent': require('./tumor/TumorPresent').default,
-    'IsPrimaryTumor': require('./tumor/IsPrimaryTumor').default,
-    'TumorDimensions': require('./tumor/TumorDimensions').default,
-    'TumorDimension1': require('./tumor/TumorDimension1').default,
-    'TumorDimension2': require('./tumor/TumorDimension2').default,
-    'TumorDimension3': require('./tumor/TumorDimension3').default,
-    'SizeOfGrossTumorBed': require('./tumor/SizeOfGrossTumorBed').default,
-    'TumorMargins': require('./tumor/TumorMargins').default,
-    'TumorMarginDescription': require('./tumor/TumorMarginDescription').default,
-    'Cellularity': require('./tumor/Cellularity').default,
-    'PercentageInSituCarcinoma': require('./tumor/PercentageInSituCarcinoma').default,
-  },
-};
 
+/**
+ * The Class Registry concept is designed to allow for users of the SHR ES6 classes to fix bugs and add extensions
+ * in a way that makes maintaining those changes across re-generations easier.
+ * Users should subclass any relevant classes, then update the class registry
+ * (in a separate location, since the file is generated)
+ * with a reference to the new class.
+ * Note: this class structure is needed to avoid circular dependency issues,
+ * as this refers to all the individual ES6 classes, and the individual ES6 classes all refer to this.
+ */
+class Registry {
+  initialize() {
+    this.internalRegistry = {
+      'shr.adverse': {
+        'Seriousness': require('./shr/adverse/Seriousness').default,
```

Comment:
Tremendous nitpick that I don't expect to be addressed immediately but it would be nice if these namespaces were ordered alphabetically.

Reply:
Updated so the namespaces are sorted, and classes within namespaces are sorted. Also submitted a PR to the es6-exporter so future versions will be sorted by default. (This was the easiest of the comments to address)